File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.js
function solution(numer1, denom1, numer2, denom2) {
    var answer = [];
    // 분모 분자 구하기
    let denominator = denom1 * denom2;
    let molecule = numer1 * denom2 + denom1 * numer2;
    // 분모와 분자의 최대공약수 구하기
    function gcd(a, b) {
        return b === 0 ? a : gcd(b , a % b);
    }
    let divisor = gcd(molecule, denominator);
    // 분모 분자를 최대공약수로 나누기 
    molecule /= divisor;
    denominator /= divisor;
    // 반환 값 넣어서 return
    answer = [molecule, denominator];
    return answer;
}

File: 프로그래머스/0/120808. 분수의 덧셈/분수의 덧셈.java
class Solution {
    public int[] solution(int numer1, int denom1, int numer2, int denom2) {
        int numerator = numer1*denom2 + numer2*denom1;
        int dominator = denom1 * denom2;
        int gcd = gcd(numerator,dominator);
        
        numerator /= gcd;
        dominator /= gcd;
        
         int[] answer = {numerator,dominator};
        return answer;
    }
        private int gcd(int a, int b){
            while(b!=0){
            int temp=b;
            b=a%b;
            a=temp;   
            }
            return a;  
    }
}